src/main/java/com/shopify/ShopifySdk.java

```diff
@@ -688,24 +712,72 @@ public ShopifyPage<ShopifyOrder> getOrders(final String pageInfo, final int page
+	 * 
+	 * @param fulfillmentOrders
+	 *            the fulfillment orders list to create the new fulfillment
+	 * @return the newly created fulfillment
+	 */
+	public ShopifyFulfillment createFulfillment(
+			final ShopifyFulfillmentCreationRequest shopifyFulfillmentCreationRequest,
+			List<ShopifyFulfillmentOrder> fulfillmentOrders) {
+		return this.createFulfillmentWithFulfillmentOrderApi(shopifyFulfillmentCreationRequest, fulfillmentOrders);
+	}
+
+	/**
+	 * Updates a fulfillment in shopify
+	 * 
+	 * @deprecated starting in March 01, 2023 this method will be obsolete
+	 * @param shopifyFulfillmentUpdateRequest
+	 *            the fulfillment update request
+	 * @return the newly updated fulfillment
+	 */
+	@Deprecated
+	public ShopifyFulfillment updateFulfillment(final ShopifyFulfillmentUpdateRequest shopifyFulfillmentUpdateRequest) {
 		final ShopifyFulfillmentRoot shopifyFulfillmentRoot = new ShopifyFulfillmentRoot();
-		final ShopifyFulfillment shopifyFulfillment = shopifyFulfillmentCreationRequest.getRequest();
-
+		final ShopifyFulfillment shopifyFulfillment = shopifyFulfillmentUpdateRequest.getRequest();
 		shopifyFulfillmentRoot.setFulfillment(shopifyFulfillment);
-		final Response response = post(buildOrdersEndpoint().path(shopifyFulfillment.getOrderId()).path(FULFILLMENTS),
-				shopifyFulfillmentRoot);
+		final Response response = put(buildOrdersEndpoint().path(shopifyFulfillment.getOrderId()).path(FULFILLMENTS)
+				.path(shopifyFulfillment.getId()), shopifyFulfillmentRoot);
 		final ShopifyFulfillmentRoot shopifyFulfillmentRootResponse = response.readEntity(ShopifyFulfillmentRoot.class);
 		return shopifyFulfillmentRootResponse.getFulfillment();
 	}
 
-	public ShopifyFulfillment updateFulfillment(final ShopifyFulfillmentUpdateRequest shopifyFulfillmentUpdateRequest) {
-		final ShopifyFulfillmentRoot shopifyFulfillmentRoot = new ShopifyFulfillmentRoot();
+	/**
+	 * Based on the api starting from 2023-04 it only updates tracking
+	 * information
+	 * 
+	 * @param shopifyFulfillmentUpdateRequest
+	 *            the information needed to update the fulfillment's tracking
+	 *            info
+	 * @return the updated version of the shopify fulfillment
+	 */
+	public ShopifyFulfillment updateFulfillmentTrackingInfo(
+			final ShopifyFulfillmentUpdateRequest shopifyFulfillmentUpdateRequest) {
 		final ShopifyFulfillment shopifyFulfillment = shopifyFulfillmentUpdateRequest.getRequest();
-		shopifyFulfillmentRoot.setFulfillment(shopifyFulfillment);
-		final Response response = put(buildOrdersEndpoint().path(shopifyFulfillment.getOrderId()).path(FULFILLMENTS)
-				.path(shopifyFulfillment.getId()), shopifyFulfillmentRoot);
+		final ShopifyUpdateFulfillmentPayloadRoot payload = LegacyToFulfillmentOrderMapping
+				.toUpdateShopifyFulfillmentPayloadRoot(shopifyFulfillment);
+
+		final Response response = post(
+				getWebTarget().path(FULFILLMENTS).path(shopifyFulfillment.getId()).path(UPDATE_TRACKING), payload);
```

Comment:
Interesting. This endpoint isn't going away?

Reply:
ya, update fulfillment works out almost in the same way. The difference being that there is no more general PUT for updates